microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/project/AbstractConfigSource.java

```diff
@@ -149,6 +159,28 @@ public Integer getPropertyAsInt(String key) {
 		return null;
 	}
 
+	private Set<String> getPropertyKeys() {
+		T config = getConfig();
+		if (config == null) {
+			return Collections.<String>emptySet();
+		}
+		return getPropertyKeys(config);
+	}
+
+	@Override
+	public Map<String, MicroProfileConfigPropertyInformation> getPropertyInformations(String propertyKey) {
+		Map<String, MicroProfileConfigPropertyInformation> infos = new HashMap<>();
+		getPropertyKeys().stream() //

```

Comment:
@datho7561 I promise you it's my last comment. Your PR is great, but I wonder if we could avoid for yaml computing all properties keys with YamlUtils.flattenYamlMapToKeys.

My suggestion is to set getPropertyInformations as abstract in AbstractConfigSource.:

`public abstract Map<String, MicroProfileConfigPropertyInformation> getPropertyInformations(String propertyKey);`

 * for PropertiesConfigSource, you implement getPropertyInformations like you did here  since you have directly all properties.
 * for YamlConfigSource you don't compute all properties keys, but you use Yaml API to visit and search the given propertyKey. It will avoid computing all properties keys although you want just visit yaml tree structure.

I don't know if it's easy to do, but I think it will be more clean to do that.

Reply:
Okay. I'm looking into it.